include/pagmo/algorithms/bee_colony.hpp

```diff
@@ -0,0 +1,365 @@
+                }
+                sump += p[i];
+            }
+            for (decltype(NP) i = 0u; i < NP; ++i) {
+                p[i] /= sump;
+            }
+            vector_double::size_type s = 0u;
+            decltype(NP) t = 0u;
+            // for each onlooker bee
+            while (t < NP) {
+                // probabilistic selection of a food source
+                auto r = rrng(m_e);
+                if (r < p[s]) {
+                    ++t;
+                    newsol = X[s];
+                    // selects a random component of the decision vector
+                    auto comp2change = comprng(m_e);
+                    // selects a random decision vector in the population other than the current
+                    auto rdv = dvrng(m_e);
+                    if (rdv >= s) {
+                        ++rdv;
+                    }
+                    // mutate new solution
+                    newsol[comp2change] += phirng(m_e) * (newsol[comp2change] - X[rdv][comp2change]);
+                    // if the generated parameter value is out of boundaries, shift it into the boundaries
+                    if (newsol[comp2change] < lb[comp2change]) {
+                        newsol[comp2change] = lb[comp2change];
+                    }
+                    if (newsol[comp2change] > ub[comp2change]) {
+                        newsol[comp2change] = ub[comp2change];
+                    }
+                    // if the new solution is better than the old one replace it and reset its trial counter
+                    auto newfitness = prob.fitness(newsol);
+                    if (newfitness[0] < fit[s][0]) {
+                        fit[s][0] = newfitness[0];
+                        X[s][comp2change] = newsol[comp2change];
+                        pop.set_xf(s, newsol, newfitness);
+                        trial[s] = 0;
+                    } else {
+                        ++trial[s];
+                    }
+                }
+                s = (s + 1) % NP;
+            }
+            // Logs and prints (verbosity modes > 1: a line is added every m_verbosity generations)
+            if (m_verbosity > 0u) {
+                // Every m_verbosity generations print a log line
+                if (gen % m_verbosity == 1u || m_verbosity == 1u) {
+                    auto best_idx = pop.best_idx();
+                    // Every 50 lines print the column names
+                    if (count % 50u == 1u) {
+                        print("\n", std::setw(7), "Gen:", std::setw(15), "Fevals:", std::setw(15), "Current best:",
+                              std::setw(15), "Best:\n");
+                    }
+                    print(std::setw(7), gen, std::setw(15), prob.get_fevals() - fevals0, std::setw(15),
+                          pop.get_f()[best_idx][0], std::setw(15), pop.champion_f()[0], '\n');
+                    ++count;
+                    // Logs
+                    m_log.push_back(
```

Comment:
Just a small FYI, you can shorten a bit this line by using ``emplace_back()`` rather than ``push_back()``. You pass to ``emplace_back()`` the set of arguments needed to construct the ``log_line_type`` object, like this:

```c++
m_log.emplace_back(gen, prob.get_fevals() - fevals0, pop.get_f()[best_idx][0], pop.champion_f()[0]));
```